core/include/mutex.h

```diff
@@ -106,6 +106,12 @@ static inline int mutex_trylock(mutex_t *mutex)
 /**
  * @brief Locks a mutex, blocking.
  *
+ * Mutexes are served in FIFO order while respecting thread priority (i.e. a higher
+ * priority thread will be unlocked before a lower priority one even if it arrived
+ * later). As long as no thread holds the mutex indefinitely and as long as there
+ * are no deadlocks, all same-priority threads blocked on mutex_lock() should
+ * eventually unblock.
```

Comment:
I'm not sure about what the point being made in this sentence is? It sounds like you're just explaining what happens when a mutex is unlocked (i.e. waiting threads unblock)? Also, shouldn't _all_ threads blocked on the mutex unblock when it's unlocked?
(I've removed this sentence in the PR as AFAICT it doesn't add anything, please correct me if I'm wrong)